### PR TITLE
Job CLI:  create Job,  submit Job, list_templates, show_variables

### PR DESCRIPTION
**Pull request by chesterxgchen**
### Description
This PR depends on https://github.com/NVIDIA/NVFlare/pull/1923 (duplicated changes)

##### Goals:
Allow users to quickly learn job configuration and adapt user's own code to the new job configuration by providing a set of templates. 
####  *nvflare job list_templates*
```  
nvflare  job list_templates  -d integration/job_templates/
```
or 
simply 
```
nvflare  job list_templates  
```
if the job_templates is specified in hidden config or env. variable 

```
------------------------------------------------------------------------------------------------------------------------
   name            description                                                  controller type      client category     
------------------------------------------------------------------------------------------------------------------------
   cyclic_tf       NA                                                           NA                   NA                  
   sag_cross_pt    scatter & gather and cross-site validation workflow using py server               client_api          
   sag_pt          scatter & gather workflow using pytorch                      server               client_api          
   stats_df        FedStats: tabular data with pandas                           server               stats executor      
------------------------------------------------------------------------------------------------------------------------
```

####  *nvflare job create*

```
nvflare job create -j /tmp/nvflare/stats_job -w stats_df  -force -f config_fed_server.conf bins=20 -f config_fed_client.conf min_noise_level=0.3   -f meta.conf min_clients=2
```

```
---------------------------------------------------------------------------------------------------------------------------------------

  job folder: /tmp/nvflare/stats_job

---------------------------------------------------------------------------------------------------------------------------------------
  file_name                      var_name                       value                               component
---------------------------------------------------------------------------------------------------------------------------------------
  config_exchange.conf           exchange_format                pytorch
  config_exchange.conf           transfer_type                  DIFF

  config_fed_client.conf         data_path                      data.csv                            DFStatistics
  config_fed_client.conf         max_bins_percent               10                                  HistogramBinsCleanser
  config_fed_client.conf         max_noise_level                0.3                                 AddNoiseToMinMax
  config_fed_client.conf         min_count                      10                                  MinCountCleanser
  config_fed_client.conf         min_noise_level                0.3                                 AddNoiseToMinMax
  config_fed_client.conf         precision                      4
  config_fed_client.conf         result_cleanser_ids            ['min_count_cleanser', 'min_max_noi

  config_fed_server.conf         bins                           20
  config_fed_server.conf         enable_pre_run_task            False                               StatisticsController
  config_fed_server.conf         json_encoder_path              nvflare.app_common.utils.json_utils JsonStatsFileWriter
  config_fed_server.conf         output_path                    statistics/adults_stats.json        JsonStatsFileWriter
  config_fed_server.conf         precision                      4                                   StatisticsController
  config_fed_server.conf         range                          [0, 120]
  config_fed_server.conf         result_wait_timeout            10                                  StatisticsController
  config_fed_server.conf         wait_time_after_min_received   1                                   StatisticsController

  meta.conf                      app                            ['@ALL']
  meta.conf                      mandatory_clients              []
  meta.conf                      min_clients                    2
  meta.conf                      scope                          []

---------------------------------------------------------------------------------------------------------------------------------------

```

#### *nvflare job show_variables* 

```
nvflare job show_variables -j /tmp/nvflare/stats_job 
```

```
The following are the variables you can change in the template

----------------------------------------------------------------------------------------------------
                                                                                                    
    job folder: /tmp/nvflare/stats_job                                                                  
                                                                                                    
----------------------------------------------------------------------------------------------------
    file_name                           var_name                  value                    
----------------------------------------------------------------------------------------------------
    config_fed_client.conf              data_path                 data.csv
    config_fed_client.conf              max_bins_percent          10
    config_fed_client.conf              max_noise_level           0.3
    config_fed_client.conf              min_count                 10
    config_fed_client.conf              min_noise_level           0.3

    config_fed_server.conf              bins                      20
    config_fed_server.conf              enable_pre_run_task       False
    config_fed_server.conf              json_encoder_path         nvflare.app_common.utils.json_utils.ObjectEncoder
    config_fed_server.conf              output_path               statistics/adults_stats.json
    config_fed_server.conf              range                     [0, 120]

    meta.conf                           app                       ['@ALL']
    meta.conf                           mandatory_clients         []
    meta.conf                           min_clients               2
    meta.conf                           scope                     []

----------------------------------------------------------------------------------------------------

```
#### *nvflare job submit*
 * require POC or production system

```
nvflare job submit -j /tmp/nvflare/stats_job -f config_fed_server.conf bins=20 -f config_fed_client.conf min_noise_level=0.3   -f meta.conf min_clients=2
```
or simply 

```
nvflare job submit -j /tmp/nvflare/stats_job 
```
 
### Limitation Already know at this point
* site-specific configurations are not handled (such as site-1, 2, each have separate config_fed_client.json etc.)
 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [x] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Review by YuanTingHsieh — CHANGES_REQUESTED** [review]
Thanks for the PR.
Some comments and questions.

**Review by IsaacYangSLA — APPROVED** [review]
From user's point of view, this new command is useful and works well.

File: integration/job_templates/sag_cross_pt/config_fed_client.conf [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "format_version": 2,
+  "executors": [
+    {
+      "tasks": [
+        "train",
+        "submit_model"
+      ],
+      "executor": {
+        "path": "nvflare.app_common.np.np_trainer.NPTrainer",
```
Comment (holgerroth) [inline integration/job_templates/sag_cross_pt/config_fed_client.conf:10]: Why NPTrainer for pt?

**Review by holgerroth — DISMISSED** [review]
Very useful.

**Review by holgerroth — APPROVED** [review]

**Pull request merged**